src/telemetry/logs.ts

```diff
@@ -193,27 +182,29 @@ class OTLPLogQueueTransport extends TransportStream {
   readonly name = "OTLPLogQueueTransport";
   readonly otelLogger: OTelLogger;
 
-  constructor(private readonly telemetryCollector: TelemetryCollector) {
+  constructor(readonly telemetryCollector: TelemetryCollector) {
     super();
     // not sure if we need a more explicit name here
-    this.otelLogger = new OTelLoggerProvider().getLogger("default") as OTelLogger;
+    const loggerProvider = new LoggerProvider();
+    this.otelLogger = loggerProvider.getLogger("default");
+    const logRecordProcessor = {
+      forceFlush: async () => {
+        // no-op
+      },
+      onEmit(logRecord: LogRecord) {
+        telemetryCollector.push(logRecord);
+      },
+      shutdown: async () => {
+        // no-op
+      },
+    };
```

Comment:
This is a custom log processor we can given to the OTel logger. It's only job is to push log records in our collector queue.

We might want to tweak its parameters in the future.